### PR TITLE
Frontend Improvements 

### PR DESCRIPTION
**Pull request by JakeUrban**
Major Changes:
- Added SCSS and custom widget templates in place of django defaults
    - requires several new python packages, additional configuration options, and setup commands
- Changed `form_for_transaction()` return type
    - This is a "breaking" change with the past versions. It was necessary for anchors to be able to provide different guidance text and page titles per-form. We could do without this change, but I believe it is necessary for #78 as well.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -33,6 +33,7 @@ MANIFEST
 # Static files
 polaris/polaris/collectstatic/*
 !polaris/polaris/collectstatic/__init__.py
+polaris/polaris/static/base.css
```
Comment (JakeUrban) [inline .gitignore]: This file is now created via `python manage.py compilescss` so we don't want to check it into version control

File: Pipfile [hunk 2]
```diff
@@ -29,6 +29,9 @@ django-cors-headers = "*"
 toml = "*"
 pyjwt = "*"
 django-sslserver = "*"
+libsass = "*"
+django-compressor = "*"
+django-sass-processor = "*"
```
Comment (JakeUrban) [inline Pipfile]: Packages needed to compile SCSS

File: polaris/polaris/integrations/transactions.py [hunk 3]
```diff
@@ -74,7 +74,7 @@ def after_deposit(cls, transaction: Transaction):
     @classmethod
     def form_for_transaction(
         cls, transaction: Transaction
-    ) -> Optional[Type[forms.Form]]:
+    ) -> Optional[Tuple[Type[forms.Form], Dict]]:
```
[the remark this answers is not in the document]
Comment (msfeldstein) [reply polaris/polaris/integrations/transactions.py]: not a big deal at all more just curious about python

[the remark this answers is not in the document]
Comment (JakeUrban) [reply polaris/polaris/integrations/transactions.py]: We could do something like this:
```
class PolarisFormContext:
    def __init__(self, form: forms.Form, context: Dict):
        self.form = form
        self.context = context
```
Or we can remove `self.context` and just make `self.title`, `self.guidance`, `self.icon_label`, etc.

But personally I'd rather not require the user to use this object. Logically its the same, but if you think the mutli-return-value-or-none thing weird we can do this.

File: polaris/polaris/templates/base.html [hunk 4]
```diff
@@ -1,41 +1,102 @@
...
     {% block 'title' %}
-    <title>SEP24 reference server</title>
+    <title>{% if title %} {{ title|safe }} {% else %} SEP24 reference server {% endif %}</title>
     {% endblock %}
 
+    <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Cabin&display=swap">
+    <link rel="stylesheet" href="{% sass_src 'base.scss' %}" />
+    <script
+      async
+      src="https://unpkg.com/flatpickr@4.6.3/dist/flatpickr.min.js"
```
[the remark this answers is not in the document]
Comment (JakeUrban) [reply polaris/polaris/templates/base.html]: Thats probably a question for Carl. I can't seem to be able to tag him though

File: polaris/polaris/integrations/transactions.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (msfeldstein) [reply polaris/polaris/integrations/transactions.py]: no its not weird, i was thinking more about how to autocomplete what values of Dict are allowed.  Definitely not something we need for this PR

File: polaris/polaris/templates/base.html (hunk 4 above)
[the remark this answers is not in the document]
Comment (ire-and-curses) [reply polaris/polaris/templates/base.html]: >The autocomplete results are restricted to repository collaborators and any other participants on the thread.
(https://help.github.com/en/github/writing-on-github/basic-writing-and-formatting-syntax#mentioning-people-and-teams)
But you can still do @vcarl

File: polaris/polaris/integrations/transactions.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (JakeUrban) [reply polaris/polaris/integrations/transactions.py]: Oh I see. Any keys that aren't used are simply ignored

**Pull request merged**